src/libecalc/presentation/yaml/domain/time_series_collections.py

```diff
@@ -0,0 +1,52 @@
+from datetime import datetime
+from typing import Dict, List, Set
+
+from libecalc.presentation.yaml.domain.time_series import TimeSeries
+from libecalc.presentation.yaml.domain.time_series_collection import TimeSeriesCollection
+from libecalc.presentation.yaml.domain.time_series_exceptions import TimeSeriesNotFound
+from libecalc.presentation.yaml.domain.time_series_provider import TimeSeriesProvider
+from libecalc.presentation.yaml.domain.time_series_resource import TimeSeriesResource
+from libecalc.presentation.yaml.resource import Resource
+from libecalc.presentation.yaml.yaml_types.time_series.yaml_time_series import YamlTimeSeriesCollection
+
+
+class TimeSeriesCollections(TimeSeriesProvider):
```

Comment:
does it make sense that both a timeseries and a timeseries collection has the same interface? it makes e.g. reference_id a bit ambigious of what it is - that can lead to errors? also, quite many other metods are needed in each of the classes that doesnt match the interface/mixin/protocol ...

Reply:
It does not make sense, but it _could_ make sense that all domain classes should have an id for example (reference_id in this case). 

The TimeSeriesProvider is only used for TimeSeriesCollections and TimeSeriesCollection. Since both just provide time_series.

So timeseries and timeseries collection should not have the same interface. 

TimeSeriesProvider has potential IMO. As we maybe could separate ResourceService into TimeSeriesProvider and FacilityResourceService (or something). 

Provider vs Service is not something I have thought much about btw.